### PR TITLE
6501 pedagogical support form implementation

### PR DESCRIPTION
**Pull request by TeroOtavia**
Pedagogical support form for uppersecondary student. This pr includes new custom made wizard component with all its own functionalities and other subcomponetns etc. It is also used by this new form. 

Pyramus 1448 branch is needed

Resolves: #6501

### CODE REVIEW

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/pedagogy/rest/PedagogyRestService.java [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+    try {
+      SchoolDataIdentifier identifier = SchoolDataIdentifier.fromId(studentIdentifier);
+      UserContactInfo contactInfo = userController.getUserContactInfo(identifier.getDataSource(), identifier.getIdentifier());
+      model.setStudentInfo(toMap(contactInfo));
+      Long formCreator = getFormCreator(form);
+      if (formCreator != null) {
+        model.setOwnerId(formCreator);
+        UserEntity userEntity = userEntityController.findUserEntityById(formCreator);
+        if (userEntity != null) {
+          contactInfo = userController.getUserContactInfo(userEntity.getDefaultSchoolDataSource().getIdentifier(), userEntity.getDefaultIdentifier());
```
Comment (aviljakaine) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/pedagogy/rest/PedagogyRestService.java:281]: Does the form really need all this information about the creator/owner (birthday, address, phone email etc)? Especially if the creator/owner can be a staff member.

**Review by aviljakaine — CHANGES_REQUESTED** [review]
- package.json hats off
- redo package-lock.json
- check if the exposed contactinfo is appropriate

**Review by avukkonen — APPROVED** [review]

**Review by aviljakaine — DISMISSED** [review]
One question about access check, if not needed, backend is ok

**Review by aviljakaine — APPROVED** [review]

**Review by jjlankinen — APPROVED** [review]
Re-approving frontend changes on behalf of a another member

**Pull request merged**